module/zfs/ddt.c

```diff
@@ -837,6 +876,181 @@ ddt_key_compare(const void *x1, const void *x2)
 	return (TREE_ISIGN(cmp));
 }
 
+/* Create the containing dir for this DDT and bump the feature count */
+static void
+ddt_create_dir(ddt_t *ddt, dmu_tx_t *tx)
+{
+	ASSERT3U(ddt->ddt_dir_object, ==, 0);
+	ASSERT3U(ddt->ddt_version, ==, DDT_VERSION_FDT);
+
+	char name[DDT_NAMELEN];
+	snprintf(name, DDT_NAMELEN, DMU_POOL_DDT_DIR,
+	    zio_checksum_table[ddt->ddt_checksum].ci_name);
+
+	ddt->ddt_dir_object = zap_create_link(ddt->ddt_os,
+	    DMU_OTN_ZAP_METADATA, DMU_POOL_DIRECTORY_OBJECT, name, tx);
```

Comment:
This is probably irrational paranoia, but you may want to consider adding a warning:
```diff
diff --git a/module/zfs/zio_compress.c b/module/zfs/zio_compress.c
index c8a10db74..fc663ece4 100644
--- a/module/zfs/zio_compress.c
+++ b/module/zfs/zio_compress.c
@@ -48,6 +48,9 @@ static unsigned long zio_decompress_fail_fraction = 0;
 
 /*
  * Compression vectors.
+ *
+ * NOTE: DO NOT CHANGE THE NAMES OF THESE COMPRESSION FUNCTIONS.  THEY ARE USED
+ * BY FAST DDT ZAP ENTRIES.
  */
 zio_compress_info_t zio_compress_table[ZIO_COMPRESS_FUNCTIONS] = {
        {"inherit",     0,      NULL,           NULL, NULL},
```

Reply:
Done